"KUBE_COVER=y make check" fails

**Is this a request for help?**
No

**What keywords did you search in Kubernetes issues before filing this one?** 
KUBE_COVER

**Is this a BUG REPORT or FEATURE REQUEST?**:
Bug report

**Kubernetes version** (use `kubectl version`):
Client Version: version.Info{Major:"1", Minor:"4+", GitVersion:"v1.4.0-alpha.2.1926+956501b1f09237-dirty", GitCommit:"956501b1f0923786c404c062d8ba001a10b3a784", GitTreeState:"dirty", BuildDate:"2016-08-30T02:14:22Z", GoVersion:"go1.6.2", Compiler:"gc", Platform:"linux/amd64"}

master at 30 August 2016

**Environment**:
- Ubuntu 16.04

**What happened**:
Note: just showing failures, not passes.
```
$ KUBE_COVER=y make check
...
+++ [0830 17:15:05] Building go targets for linux/amd64:
	cmd/libs/go2idl/conversion-gen
E0830 17:15:08.878493   17154 conversion.go:589] Warning: could not generate autoConvert functions for k8s.io/kubernetes/pkg/apis/apps/v1alpha1.PetSetSpec <-> k8s.io/kubernetes/pkg/apis/apps.PetSetSpec
E0830 17:15:08.904578   17154 conversion.go:589] Warning: could not generate autoConvert functions for k8s.io/kubernetes/pkg/apis/extensions/v1beta1.HorizontalPodAutoscalerSpec <-> k8s.io/kubernetes/pkg/apis/autoscaling.HorizontalPodAutoscalerSpec
E0830 17:15:08.911983   17154 conversion.go:589] Warning: could not generate autoConvert functions for k8s.io/kubernetes/pkg/apis/extensions/v1beta1.JobSpec <-> k8s.io/kubernetes/pkg/apis/batch.JobSpec
E0830 17:15:08.924136   17154 conversion.go:589] Warning: could not generate autoConvert functions for k8s.io/kubernetes/pkg/apis/extensions/v1beta1.RollingUpdateDeployment <-> k8s.io/kubernetes/pkg/apis/extensions.RollingUpdateDeployment
E0830 17:15:08.926271   17154 conversion.go:589] Warning: could not generate autoConvert functions for k8s.io/kubernetes/pkg/apis/extensions/v1beta1.ScaleStatus <-> k8s.io/kubernetes/pkg/apis/extensions.ScaleStatus
...
ok  	k8s.io/kubernetes/pkg/client/transport	0.035s	coverage: 51.7% of statements
# cover k8s.io/kubernetes/pkg/client/restclient
cover: internal error: block 268 overlaps block 270
	/home/angus/work/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/client/restclient/request.go:#27803,#27927 /home/angus/work/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/client/restclient/request.go:#27848,#27867
ok  	k8s.io/kubernetes/pkg/client/restclient	0.051s	coverage: 69.5% of statements
...
ok  	k8s.io/kubernetes/pkg/registry/secret	0.122s	coverage: 30.8% of statements
--- FAIL: TestServiceRegistryExternalTrafficAnnotationHealthCheckNodePortUserAllocation (0.00s)
	rest_test.go:1012: Unexpected failure creating service Internal error occurred: Failed to allocate requested HealthCheck nodePort 30200: provided port is already allocated
panic: interface conversion: runtime.Object is nil, not *api.Service [recovered]
	panic: interface conversion: runtime.Object is nil, not *api.Service

goroutine 105 [running]:
panic(0x19c3ac0, 0xc820014ec0)
	/usr/lib/go-1.6/src/runtime/panic.go:481 +0x3e6
testing.tRunner.func1(0xc82008ae10)
	/usr/lib/go-1.6/src/testing/testing.go:467 +0x192
panic(0x19c3ac0, 0xc820014ec0)
	/usr/lib/go-1.6/src/runtime/panic.go:443 +0x4e9
k8s.io/kubernetes/pkg/registry/service.TestServiceRegistryExternalTrafficAnnotationHealthCheckNodePortUserAllocation(0xc82008ae10)
	/home/angus/work/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/registry/service/rest_test.go:1014 +0x87d
testing.tRunner(0xc82008ae10, 0x272d928)
	/usr/lib/go-1.6/src/testing/testing.go:473 +0x98
created by testing.RunTests
	/usr/lib/go-1.6/src/testing/testing.go:582 +0x892
FAIL	k8s.io/kubernetes/pkg/registry/service	0.071s
ok  	k8s.io/kubernetes/pkg/registry/service/allocator	0.002s	coverage: 93.3% of statements
...
ok  	k8s.io/kubernetes/vendor/k8s.io/client-go/1.4/discovery	0.021s	coverage: 52.0% of statements
# cover k8s.io/kubernetes/vendor/k8s.io/client-go/1.4/rest
cover: internal error: block 264 overlaps block 266
	/home/angus/work/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/k8s.io/client-go/1.4/rest/request.go:#27655,#27779 /home/angus/work/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/k8s.io/client-go/1.4/rest/request.go:#27700,#27719
ok  	k8s.io/kubernetes/vendor/k8s.io/client-go/1.4/dynamic	0.041s	coverage: 53.2% of statements
..
ok  	k8s.io/kubernetes/vendor/k8s.io/client-go/1.4/tools/record	0.048s	coverage: 88.5% of statements
+++ [0830 18:35:03] Combined coverage report: /tmp/k8s_coverage/v1,apps/v1alpha1,authentication.k8s.io/v1beta1,authorization.k8s.io/v1beta1,autoscaling/v1,batch/v1,batch/v2alpha1,certificates/v1alpha1,extensions/v1beta1,federation/v1beta1,policy/v1alpha1,rbac.authorization.k8s.io/v1alpha1,imagepolicy.k8s.io/v1alpha1/20160830-171509/combined-coverage.html
Makefile:118: recipe for target 'check' failed
make: *** [check] Error 1
```

**What you expected to happen**:
No errors (note `make check` passed).

**How to reproduce it** (as minimally and precisely as possible):
$ KUBE_COVER=y make check

**Anything else do we need to know**:
I am on Ubuntu 10.04

--- a/hack/make-rules/test.sh
+++ b/hack/make-rules/test.sh
@@ -224,13 +224,21 @@ runTests() {
   # must make sure the output from PARALLEL runs is not mixed. To achieve this,
   # we spawn a subshell for each PARALLEL process, redirecting the output to
   # separate files.
-  # cmd/libs/go2idl/generator is fragile when run under coverage, so ignore it for now.
-  # see: https://github.com/kubernetes/kubernetes/issues/24967
+
+  # ignore paths:
+  # cmd/libs/go2idl/generator: is fragile when run under coverage, so ignore it for now.
+  #                            https://github.com/kubernetes/kubernetes/issues/24967
+  # vendor/k8s.io/client-go/1.4/rest: causes cover internal errors
+  #                            https://github.com/golang/go/issues/16540
+  cover_ignore_dirs="cmd/libs/go2idl/generator|vendor/k8s.io/client-go/1.4/rest"
+  for path in $(echo $cover_ignore_dirs | sed 's/|/ /g'); do
+      echo -e "skipped\tk8s.io/kubernetes/$path"
+  done
   #
   # `go test` does not install the things it builds. `go test -i` installs
   # the build artifacts but doesn't run the tests.  The two together provide
   # a large speedup for tests that do not need to be rebuilt.
-  printf "%s\n" "${@}" | grep -v "cmd/libs/go2idl/generator"| xargs -I{} -n1 -P${KUBE_COVERPROCS} \
+  printf "%s\n" "${@}" | grep -Ev $cover_ignore_dirs | xargs -I{} -n1 -P${KUBE_COVERPROCS} \
     bash -c "set -o pipefail; _pkg=\"{}\"; _pkg_out=\${_pkg//\//_}; \
         go test -i ${goflags[@]:+${goflags[@]}} \
           ${KUBE_RACE} \

--- a/pkg/client/restclient/request.go
+++ b/pkg/client/restclient/request.go
@@ -901,10 +901,11 @@ func (r *Request) transformResponse(resp *http.Response, req *http.Request) Resu
 	}
 
 	if glog.V(8) {
-		switch {
-		case bytes.IndexFunc(body, func(r rune) bool { return r < 0x0a }) != -1:
+		if bytes.IndexFunc(body, func(r rune) bool {
+			return r < 0x0a
+		}) != -1 {
 			glog.Infof("Response Body:\n%s", hex.Dump(body))
-		default:
+		} else {
 			glog.Infof("Response Body: %s", string(body))
 		}
 	}

